fl4health/model_bases/autoencoders_base.py

```diff
@@ -0,0 +1,131 @@
+from abc import ABC, abstractmethod
+from typing import Callable, Optional, Tuple
+
+import torch
+import torch.nn as nn
+
+
+class AbstractAe(nn.Module, ABC):
+    def __init__(
+        self,
+        encoder: nn.Module,
+        decoder: nn.Module,
+    ) -> None:
+        """
+        The base class for all autoencoder based models.
+        All we need to define such model is the type of the model, and the structure
```

Comment:
I think the first part of this comment "All we need to define such model is the type of the model..." is from a previous iteration. Perhaps we can just start with "All we need to define such a model is the structure of the encoder..."